fastlane/Fastfile

```diff
@@ -591,12 +591,21 @@ private_lane :ensure_tool_name_formatting do
     Dir["**/*.md"].each do |path|
       content = File.read(path)
       Fastlane::TOOLS.each do |tool|
-        errors << "Use _#{tool}_ instead of `#{tool}` to mention a tool in the docs in '#{path}'" if content.include?("`#{tool}`")
-        errors << "Use _#{tool}_ instead of `_#{tool}_` to mention a tool in the docs in '#{path}'" if content.include?("`_#{tool}_`")
-        errors << "Use [_#{tool}_] instead of [#{tool}] to mention a tool in the docs in '#{path}'" if content.include?("[#{tool}]")
-        errors << "Use <em>#{tool}<em> instead of <code>#{tool}</code> to mention a tool in the docs in '#{path}'" if content.include?("<code>#{tool}</code>")
-        if content.include?("_#{tool.to_s.capitalize}_") || content.include?("`#{tool.to_s.capitalize}`")
-          errors << "fastlane tools have to be formatted in lower case: #{tool} in '#{path}'"
+        lines = content.split("\n")
+        lines.each_with_index do |line, index|
+          line_number = index + 1
+          # Ignore checks if line starts with "^\s*- ❌" (i.e. if it's a line that's already been marked as incorrect)
+          # and if the file is CONTRIBUTING.md (i.e. if it's the branding guidelines section)
+          next if line =~ /^\s*- ❌/ && path == "CONTRIBUTING.md"
+          errors << "Use _#{tool}_ instead of `#{tool}` to mention a tool in the docs in '#{path}:#{line_number}'" if line.include?("`#{tool}`")
+          errors << "Use _#{tool}_ instead of `_#{tool}_` to mention a tool in the docs in '#{path}:#{line_number}'" if line.include?("`_#{tool}_`")
+          errors << "Use [_#{tool}_] instead of [#{tool}] to mention a tool in the docs in '#{path}:#{line_number}'" if line.include?("[#{tool}]")
+          errors << "Use _#{tool}_ instead of **#{tool}** to mention a tool in the docs in '#{path}:#{line_number}'" if line.include?("**#{tool}**")
+          errors << "Use <em>#{tool}<em> instead of <code>#{tool}</code> to mention a tool in the docs in '#{path}:#{line_number}'" if line.include?("<code>#{tool}</code>")
+          errors << "Use <em>#{tool}<em> or _#{tool}_ instead of <ins>#{tool}</ins> to mention a tool in the docs in '#{path}:#{line_number}'" if line.include?("<ins>#{tool}</ins>")
+          if line.include?("_#{tool.to_s.capitalize}_") || line.include?("`#{tool.to_s.capitalize}`") || line.include?(" #{tool.to_s.capitalize} ")
```

Comment:
Note that this test only checks if the tool name had its first letter capitalized (e.g. `Fastlane`) but not any other case variation. So it wouldn't catch typos like `FastLane`.

It might be better to use `line.scan(/\b#{tool}\b/i)` to capture each occurrences of the tool name used as a word (`\b` = "word boundary" marker in Regexp) in a case-insensitive way, then for each occurrence found by `scan`, check if it's == to `to_s.downcase` or not.

Something like (not tested):
```ruby
line.scan(/\b#{Regexp.escape(tool.to_s)}\b./) do |match|
  errors << "fastlane tools have to be formatted in lowercase: #{tool} in '#{path}:#{line_number}'" unless match == match.downcase
end

Reply:
I tried using `\b` before but it doesn't work, it catches more false positives, e.g. `app = Spaceship::ConnectAPI.get_app(app_id: 1013943394).first` <-- "Spaceship" is caught here. Too troublesome to workaround those 🤷  that's why I kept those silly (yet simple) delimiters hehe

I just tried your suggested approach using delimiters and there are a few false positives:

https://github.com/fastlane/fastlane/blob/ab9fb3249adaddf6c05655891b4d280415dfc84e/RespondingToIssuesAndPullRequests.md?plain=1#L66

https://github.com/fastlane/fastlane/blob/ab9fb3249adaddf6c05655891b4d280415dfc84e/Testing.md?plain=1#L86

https://github.com/fastlane/fastlane/blob/ab9fb3249adaddf6c05655891b4d280415dfc84e/Testing.md?plain=1#L91

What do you think we should do? Renaming those env vars to temporarily get rid of the false positive could be an option, but then no one in the future would be able to name an env var that has "FASTLANE" in it 😬 ?

Reply:
The algo ended up like this btw:

```ruby
# Check if the tool is mentioned in the line, but not in lowercase
delimiters = ["_", " ", "`", "\""]
delimiters.each do |delimiter|
  line.scan(/#{delimiter}#{Regexp.escape(tool.to_s)}#{delimiter}/i) do |match|
    errors << "fastlane tools have to be formatted in lowercase: #{tool} in '#{path}:#{line_number}'" unless match == match.downcase
  end
end
```

Reply:
I could also exclude full caps `FASTLANE` somehow lol WDYT @AliSoftware?

Reply:
I don't think we should exclude all-capts `FASTLANE`.

For your code above, you could have gone with the list of delimiters directly in the `Regexp`, using:
```
line.scan(/([_ `"])#{Regexp.escape(tool.to_s)}\1/i) do |match|`
```
and thus completely avoid doing the loop on `delimiter`

Reply:
😲 Great suggestion! I'll apply that change :) 

How do you suggest we solve those 3 false positives? 👀

Reply:
> How do you suggest we solve those 3 false positives?

We can allow `[A-Z_]+FASTLANE` and `FASTLANE_[A-Z_]+` as exceptions then, to allow it only if the name is part of a "upper-snake-case" word (like for an env var name)

Reply:
<img width="1091" alt="image" src="https://github.com/fastlane/fastlane/assets/8419048/7d6c3762-0e68-4c7c-b16e-ade2d68ef78f">


Hmm I came up with a different regex than yours, lemme check yours (as it's simpler)

Reply:
Yours works and it's simpler, check out the diff:

```
looks_like_an_env_var = line.match?(/[\`"][\w<_"]+#{matched_tool}[\w>_"]+[\`"]/)
looks_like_an_env_var = line.match?(/[A-Z_]+_#{matched_tool}_[A-Z_]+/)
```

Mine was checking if it was wrapped around backticks, double quotes, etc. That's probably not needed 😛 plus mine was allowing lowercase letters, we don't want those either...

Reply:
Ready for review again @AliSoftware 😊